### ISSUE TITLE
Please add a Option to use rar DISK on replace a failed one

### ISSUE THREAD

**Issue by arogarth** [openzfs/zfs#14174]
### Describe the feature would like to see added to OpenZFS

We have a ZFS created with FreeBSD years ago and migrate the machine to Ubuntu with OpenZFS - all works well.
Now some disks failed and we cannot replace them... WHY?

Creating the zfs with FreeBSD the whole raw disk was used as a device without partitioning. On replacing the DIsk with OpenZFS it is creating a partition table. Now the 8M at the end are missing for compatibility to the old setup.

This feature should only be used on replacing disks... And yes we know that the sectors should be equal or greater than the old failed one.

### How will this feature improve OpenZFS?

Better compatibility to old ZFS created with BSD systems

### Additional context

-

**Comment by freqlabs** [comment]
I agree, we need way to opt out of automatic partitioning on Linux, but I'm not sure how to do it. Maybe a tunable would work as a bare minimum.

**Comment by arogarth** [comment]
On replace - check the original disk - zdb list the informations of the partitioning... if the disk wasnt splitted into partition - don't do it and add the whole disk...

**Comment by scineram** [comment]
Remove partitioning crap from Linux. As @amotin suggested long ago.

**Comment by jumbi77** [comment]
+1 for that.
FYI: Referencing https://github.com/openzfs/zfs/pull/6277 https://github.com/openzfs/zfs/issues/3452 https://github.com/openzfs/zfs/issues/94

And also in a broader context https://github.com/openzfs/zfs/pull/11029 https://github.com/openzfs/zfs/issues/11408

**Comment by shodanshok** [comment]
@scineram I agree with you, but an option to opt in/out partition creation would be very useful especially when replacing a disk (if autodetection as suggested by @arogarth fails).

**Comment by pepa65** [comment]
As a (hopefully temporary) workaround:
https://github.com/openzfs/zfs/issues/94#issuecomment-405137312

**Comment by ehem** [comment]
Since I came up with the workaround almost 8 years ago, we're out of temporary.  Perhaps once a month we simply need to add comments about "rip it out".

_Rip it out!_
**Rip it out!
_Rip it out!_**

**Comment by Anuskuss** [comment]
@ehem What does your `zdb` say? Specifically `devid`, is it `xvdk1` or `xvdk`? And `whole_disk` is presumably `0`? This is what mine says (where I selected the *whole disk* but ZFS on Linux unfortunately created 2 partitions instead): 

```
children[0]:
    type: 'disk'
    id: 0
    guid: #
    path: '/dev/disk/by-id/ata-#-part1'
    devid: 'ata-#-part1'
    phys_path: 'pci-0000:00:17.0-ata-1.0'
    whole_disk: 1
```

I'd be interested in converting my existing pool to partition-less. Probably just `wipefs`ing all drives, maybe even `dd`ing (some amount of data at the start of) `sda1` to `sda` (then `zfs scrub`ing afterwards). But what's the point if this workaround is just a hack? I still think I'd want `whole_disk=1`. I don't know how much it really does but one thing I can think of is that ZFS enables the write cache if it knows it "owns" the whole drive.

There has been plenty of issues/FRs/PRs opened for this particual annoyance, but most have been closed. AFAICT this and #3452 are the only ones left. So I guess it's fair to say that this won't be happening (properly anyway).
